src/components/LogRow/BaseRow/index.tsx

```diff
@@ -114,27 +118,40 @@ interface ProcessedBaseRowProps {
   children: string;
   searchTerm?: RegExp;
   color?: string;
+  highlights?: RegExp;
   ["data-cy"]?: string;
 }
 
 const ProcessedBaseRow: React.FC<ProcessedBaseRowProps> = memo((props) => {
-  const { children, searchTerm, color, "data-cy": dataCy } = props;
+  const { children, searchTerm, color, "data-cy": dataCy, highlights } = props;
   const memoizedLogLine = useMemo(() => {
     let render = children;
     if (searchTerm) {
       // escape the matching string to prevent XSS
-      render = render.replace(
-        searchTerm,
+      render = render.replaceAll(
+        new RegExp(searchTerm, "g"),
         (match) => `<mark>${escapeHtml(match)}</mark>`
       );
     }
+    if (highlights) {
+      const shouldCheckForOverlappingRegex = searchTerm !== undefined;
+      const hasOverlappingRegexes =
+        shouldCheckForOverlappingRegex &&
+        hasOverlappingRegex(searchTerm, highlights, children);
+      if (!hasOverlappingRegexes) {
+        render = render.replaceAll(
+          new RegExp(highlights, "g"),
+          (match) => `<mark>${escapeHtml(match)}</mark>`
+        );
+      }
+    }
```

Comment:
this is possibly a really dumb comment lol... but I just wanted to leave it as food for thought. I was just trying to see if we could reduce some of the complicated logic. This passes all of the tests written for `BaseRow`.
```suggestion
    // I stole this expression from spruce
    const impossibleRegex = /^\b$/;

    const searchTermRegex = searchTerm
      ? new RegExp(searchTerm)
      : impossibleRegex;

    const highlightRegex = highlights
      ? new RegExp(highlights)
      : impossibleRegex;

    // Add "i" if case is insensitive. I am assuming the case applies to both search and highlights since highlights can no longer manage their own case sensitivity. However if this assumption doesn't hold then this solution is meaningless.
    const replaceRegex = new RegExp(
      `${searchTermRegex.source}|${highlightRegex.source}`,
      "gi"
    );

    const render = children.replace(
      replaceRegex,
      (match) => `<mark>${escapeHtml(match)}</mark>`
    );
```

Reply:
Good catch i fixed the sensitivity bug!